src/Microsoft.AspNet.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -265,24 +264,50 @@ private static bool CanUpdateReadOnlyProperty(Type propertyType)
             return true;
         }
 
-        private async Task<ModelBindingResult> CreateAndPopulateDto(
+        // Returned dictionary contains entries corresponding to properties against which binding was attempted. If
+        // binding failed, the entry's value will have IsModelSet == false. Binding is attempted for all elements of
+        // propertyMetadatas.
+        private async Task<IDictionary<ModelMetadata, ModelBindingResult>> BindPropertiesAsync(
             ModelBindingContext bindingContext,
             IEnumerable<ModelMetadata> propertyMetadatas)
         {
-            // create a DTO and call into the DTO binder
-            var dto = new ComplexModelDto(bindingContext.ModelMetadata, propertyMetadatas);
-
-            var metadataProvider = bindingContext.OperationBindingContext.MetadataProvider;
-            var dtoMetadata = metadataProvider.GetMetadataForType(typeof(ComplexModelDto));
+            var results = new Dictionary<ModelMetadata, ModelBindingResult>();
+            foreach (var propertyMetadata in propertyMetadatas)
+            {
+                var propertyModelName = ModelNames.CreatePropertyModelName(
+                    bindingContext.ModelName,
+                    propertyMetadata.BinderModelName ?? propertyMetadata.PropertyName);
+                var childContext = ModelBindingContext.GetChildModelBindingContext(
+                    bindingContext,
+                    propertyModelName,
+                    propertyMetadata);
+
+                // ModelBindingContext.Model property values may be non-null when invoked via TryUpdateModel(). Pass
+                // complex (including collection) values down so that binding system does not unnecessarily recreate
+                // instances or overwrite inner properties that are not bound. No need for this with simple values
+                // because they will be overwritten if binding succeeds. Arrays are never reused because they cannot
+                // be resized.
+                //
+                // ModelMetadata.PropertyGetter is not null safe; use it only if Model is non-null.
+                if (bindingContext.Model != null &&
+                    propertyMetadata.PropertyGetter != null &&
+                    propertyMetadata.IsComplexType &&
```

Comment:
`IsComplexType` just checks for a conversion from a `string`. So it includes `IsCollectionType`.